Spravo/Services/FirebaseAgent.swift

```diff
@@ -73,36 +89,68 @@ class FirebaseAgent: FirebaseAgentType {
         }
     }
     
-    func saveNewContact(userFbId: String, contact: AddressBookModel) {
-        var ref: DocumentReference? = nil
-        guard let dict = contact.dictionary else { return }
-        ref = firestore.collection("user\(userFbId)").addDocument(data: dict)
-        { error in
-            if let error = error {
-                debugPrint("Error adding document: \(error)")
-            } else {
-                debugPrint("Document added with ID: \(ref!.documentID)")
+    func saveNewContact(userFbId: String, contact: Contact, userProfileImage: UIImage? = nil, completion: @escaping (_ error: String?) -> ()) {
```

Comment:
Error handling look odd. It is just too much of it (it takes to much space, like 70% of method body)